src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,60 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(String operation, String fruit, String quantity) {
+        this.operation = Operation.of(operation);
+        this.fruit = fruit;
+        this.quantity = Integer.parseInt(quantity);
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE("b"),
+        SUPPLY("s"),
+        PURCHASE("p"),
+        RETURN("r");
+
+        private String code;
+
+        Operation(String code) {
+            this.code = code;
+        }
+
+        public String getCode() {
+            return code;
+        }
+
+        public static Operation of(String code) {
+            for (Operation operation : Operation.values()) {
+                if (operation.getCode().equals(code)) {
+                    return operation;
+                }
+            }
+            throw new IllegalArgumentException("Invalid Operation code: " + code);
```

Comment:
Throwing a raw IllegalArgumentException is fine, but it would be more user-friendly to wrap this in a custom exception that provides more context about the failure, especially since this is part of a domain-specific operation set.

Reply:
--